### PR TITLE
 Convert hui-state-icon-element to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**
Dependent on https://github.com/home-assistant/home-assistant-polymer/pull/1871

Will rebase after 1871 merged

### CODE REVIEW

**Comment by iantrich** [comment]
Rebase complete

File: src/panels/lovelace/elements/hui-state-icon-element.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  public setConfig(config: LovelaceElementConfig): void {
+    if (!config.entity) {
+      throw Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass!.states[this._config!.entity!]) {
```
Comment (zsarnett) [inline src/panels/lovelace/elements/hui-state-icon-element.ts]: Shouldnt need the `this._config!` with the `!` here and check for hass here as well

File: src/panels/lovelace/elements/hui-state-icon-element.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    const state = this.hass!.states[this._config!.entity!];
+    return html`
+      ${this.renderStyle()}
+      <state-badge
+        .stateObj=${state}
+        .title="${computeTooltip(this.hass!, this._config)}"
+        @ha-click="${() => handleClick(this, this.hass!, this._config!, false)}"
+        @ha-hold="${() => handleClick(this, this.hass!, this._config!, true)}"
```
Comment (zsarnett) [inline src/panels/lovelace/elements/hui-state-icon-element.ts]: After checking for hass take the `!` out

**Review by zsarnett — APPROVED** [review]
After changes made. Approved

File: src/panels/lovelace/elements/hui-state-icon-element.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  public setConfig(config: LovelaceElementConfig): void {
+    if (!config.entity) {
+      throw Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass!.states[this._config.entity!]) {
```
Comment (balloob) [inline src/panels/lovelace/elements/hui-state-icon-element.ts]: This is ok for now, but we really should have it render a yellow warning that entity doesn't exist.

File: src/panels/lovelace/elements/hui-state-icon-element.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import "../../../components/entity/state-badge.js";
+
+import { computeTooltip } from "../../../common/string/compute-tooltip";
```
Comment (balloob) [inline src/panels/lovelace/elements/hui-state-icon-element.ts]: Compute tooltip has to be moved inside lovelace elements folder. It is super lovelace specific, nothing common about it.

File: src/panels/lovelace/elements/hui-state-icon-element.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import "../../../components/entity/state-badge.js";
+
+import { computeTooltip } from "../../../common/string/compute-tooltip";
+import { handleClick } from "../../../common/dom/handle-click";
```
Comment (balloob) [inline src/panels/lovelace/elements/hui-state-icon-element.ts]: Also has to be moved inside the lovelace element folder.

File: src/panels/lovelace/elements/hui-state-icon-element.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+import "../../../components/entity/state-badge.js";
+
+import { computeTooltip } from "../../../common/string/compute-tooltip";
+import { handleClick } from "../../../common/dom/handle-click";
+import { longPress } from "../common/directives/long-press-directive";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceElement, LovelaceElementConfig } from "./types.js";
+import { HomeAssistant } from "../../../types.js";
+import { TemplateResult } from "lit-html";
```
Comment (balloob) [inline src/panels/lovelace/elements/hui-state-icon-element.ts]: We should have absolute imports above relative imports.

File: src/panels/lovelace/elements/hui-state-icon-element.ts (hunk 4 above)
Comment (iantrich) [reply src/panels/lovelace/elements/hui-state-icon-element.ts]: I'll note that for a future PR

**Review by balloob — APPROVED** [review]

**Pull request merged**